hypothesis-python/tests/datetime/test_zoneinfo_timezones.py

```diff
@@ -0,0 +1,65 @@
+# This file is part of Hypothesis, which may be found at
+# https://github.com/HypothesisWorks/hypothesis/
+#
+# Most of this work is copyright (C) 2013-2020 David R. MacIver
+# (david@drmaciver.com), but it contains contributions by others. See
+# CONTRIBUTING.rst for a full list of people who may hold copyright, and
+# consult the git log if you need to determine who owns an individual
+# contribution.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public License,
+# v. 2.0. If a copy of the MPL was not distributed with this file, You can
+# obtain one at https://mozilla.org/MPL/2.0/.
+#
+# END HEADER
+
+import pytest
+
+from hypothesis import given, strategies as st
+from hypothesis.errors import InvalidArgument
+from hypothesis.internal.compat import zoneinfo
+from tests.common.debug import assert_no_examples, find_any, minimal
+
+
+def test_utc_is_minimal():
+    assert minimal(st.timezones()) is zoneinfo.ZoneInfo("UTC")
+
+
+def test_can_generate_non_utc():
+    find_any(
+        st.datetimes(timezones=st.timezones()).filter(lambda d: d.tzinfo.key != "UTC")
+    )
+
+
+@given(st.data(), st.datetimes(), st.datetimes())
+def test_datetimes_stay_within_naive_bounds(data, lo, hi):
+    if lo > hi:
+        lo, hi = hi, lo
+    out = data.draw(st.datetimes(lo, hi, timezones=st.timezones()))
+    assert lo <= out.replace(tzinfo=None) <= hi
+
+
+@pytest.mark.parametrize("kwargs", [{"no_cache": 1}])
+def test_timezones_argument_validation(kwargs):
+    with pytest.raises(InvalidArgument):
+        st.timezones(**kwargs).validate()
+
+
+@pytest.mark.parametrize(
+    "kwargs",
+    [
+        # {"allow_alias": 1},
+        # {"allow_deprecated": 1},
+        {"allow_prefix": 1},
+    ],
+)
+def test_timezone_keys_argument_validation(kwargs):
+    with pytest.raises(InvalidArgument):
+        st.timezone_keys(**kwargs).validate()
+
+
+def test_can_disallow_prefixes():
+    assert_no_examples(
+        st.timezone_keys(allow_prefix=False),
+        lambda s: s.startswith(("posix/", "right/")),
+    )
```

Comment:
Is it worth adding a positive check here that `allow_prefix=True` does generate prefixed examples?

Reply:
In principle yes, but in practice the prefixed forms are only available on certain non-OSX Unixes (ie Linux or \*BSD distros) and I couldn't work out a reasonably general way to check that without making the test either tautological, or occasionally fail for the wrong reasons 😕

Reply:
Hmm. It may be worth limiting such a test to just the platform(s) known to definitely have the prefixed forms, and skipping others?

That way, it will at least catch any future change or refactor that accidentally disables generating the prefixed forms?

Reply:
Unfortunately I don't know what platforms definitely have the prefixed forms!

Our tests would pass in CI, but might fail downstream for non-ubuntu distro maintainers... having said all that, I'm just going to add the test and a big explanatory comment so it's obviously OK to patch out if it fails.

Reply:
That sounds good!